VideoPress: add notice for media-new.php

dbtlr opened the pull request:
Currently, uploading to VideoPress is a tad confusing since you cannot upload via the media-new.php, as this is an older uploader.

This adds a notice to that page informing users where to go.

<img width="1049" alt="screen shot 2016-12-15 at 1 38 28 pm" src="https://cloud.githubusercontent.com/assets/195095/21239286/d5f62e8e-c2cb-11e6-9b99-def048847304.png">

## Testing

1) Add Jetpack Professional
2) Go to Media -> Add New
3) Verify the message is there
4) Make sure it isn't appearing in other places in the admin

beaulebens commented on modules/videopress/class.jetpack-videopress.php:
```diff
@@ -109,6 +110,21 @@ public function is_connection_owner( $user_id = false ) {
+
+        ?>
+        <div id="message" class="notice notice-warning">
+            <p></p><?php echo sprintf( __( 'If you would like to upload a video to VideoPress, please use the <a href="%s">Media Library</a> instead.', 'jetpack' ), '/wp-admin/upload.php' ); ?></p>
```
Some weird `<p>` tags going on at the start of this line.

Also, you can't assume that wp-admin is in the root of a domain like that, you should use `admin_url()` to let WP handle it correctly.

beaulebens replied:
Also, rather than talk about "VideoPress", let's keep this generic and make the message something like:

"You can upload video via your [Media Library], or when creating a [new post]."

dbtlr replied:
doh! fixed!

beaulebens commented on modules/videopress/class.jetpack-videopress.php:
```diff
@@ -109,6 +110,21 @@ public function is_connection_owner( $user_id = false ) {
 	}
 
+	public function add_media_new_notice()
+    {
```
This should be at the end of the line above :)

dbtlr replied:
yep, not sure how I missed that

beaulebens commented on modules/videopress/class.jetpack-videopress.php:
```diff
@@ -109,6 +110,21 @@ public function is_connection_owner( $user_id = false ) {
+
+	    if ( $pagenow != 'media-new.php' ) {
+	        return;
+        }
```
Weird indent?

dbtlr replied:
yeah, my editor is using spaces for some reason, not sure why ...

dbtlr commented:
I made the changes you proposed @beaulebens! Thanks!

beaulebens commented:
Nice, this is looking much better. 🚢

thingalon commented:
Tested this PR after running into [this very problem](https://dopsp2.wordpress.com/2016/12/14/jetpack-4-5-is-ready-for-testing/#comment-4973). Can confirm it works.

dbtlr commented:
Given the sheer number of people trying to upload through this method and having problems, tells me I'm going to need to figure out a way to override this older uploader...

jeherve commented:
Related: #5899

dereksmart commented:
Let's get a design eye on this.  Do we want a Jetpack logo in there? 

As it is now we're not explaining _why_ someone would want to upload via media lib or new post?  If I didn't know better, this notice could be a general informational, undismissable WP notice coming from Core.  

@beaulebens @jeffgolenski 

![screen shot 2016-12-16 at 12 31 22 pm](https://cloud.githubusercontent.com/assets/7129409/21272099/c0c8e112-c38b-11e6-81ee-a83dcf7917e8.png)

dbtlr commented:
Fantastic idea to get a smarter design on this.

As it stands, the whole idea of VideoPress not working here is a bit tough to understand, even for us. Would be good to make it as plain as possible for people why the other page is a better alternative.

We could also just troll everyone with an injected Javascript redirect to the Media Library page :-P

jeffgolenski commented:
@dereksmart Good call derek.

@dbtlr It would probably be wise to include the jetpack logo in this message, like we do with our JITMS (just in time message prompts) so users don't get confused and think this is entirely related to Core. (you can just steal the code from a JITM within the code)

jeffgolenski commented:
Also, as @dereksmart mentions above, the new language changes are far less informational than the initial language. The new language looks as though it's strictly a helpful tip and makes no mention of VideoPress uploads not working from this view.

beaulebens commented:
> makes no mention of VideoPress uploads not working from this view.
Yeah I was probably overzealous, but was trying to get away from using "VideoPress" specifically. We could just make it say "Jetpack Video" instead, or something along those lines. The VideoPress brand/name isn't relevant, so I don't want to scatter that everywhere through wp-admin.

dereksmart commented:
@jeffgolenski thanks for the review, is adding the logo and changing the text something you can do?

dbtlr commented:
If I can get an example of what this should look like and if someone has a suggestion for better text, I'm more than happy to incorporate the changes.

ashleighaxios commented:
@dbtlr, thank you. Please repurpose the just in time message (JITM) code for design. If that doesn't work for any reason, let us know.
> "you can just steal the code from a JITM within the code"

For language: "If you would like to upload a video to Jetpack Video, please use the [Media Library](url) instead."

jeffgolenski commented:
@ashleighaxios Thanks for jumping in!

dbtlr commented:
I've updated the messaging to use the same style as the rest of the JITM messaging system. 

I also updated the messaging just a little bit to let the user know what the point of the message is. Open to possible revisions to the copy:

`Only videos uploaded from your <a href="%s">media library</a> or while creating a <a href="%s">new post</a> will be fully hosted by WordPress.com.`

dbtlr commented:
Also, the build failures appear to be related to some mysqli() issue that is not related to this code. I saw it in another PR as well. Not sure if it related to having rebased master.

dereksmart commented:
@jeffgolenski this is what it looks like now: 

![screen shot 2017-01-03 at 9 39 27 am](https://cloud.githubusercontent.com/assets/7129409/21610908/94fca472-d198-11e6-8f39-f80744c2f8ba.png)

What say you?

jeffgolenski commented:
Thanks for the Language changes Derek and Drew. IPF!

dereksmart commented:
Functions properly 👍

dereksmart commented:
merged to 4.5 99b7536

druesome commented:
My apologies in advance if I'm looking at this the wrong way, but I assume this has already been merged.  However, I'm not seeing this notice show up on any of my AT or self-hosted sites.  Has this been removed?  This will be very helpful while #7073 is being addressed.

zinigor commented:
@withinboredom can you please take a look? I can't see the just-in-time message anymore as well, maybe we need to move this one to the new system in order for it to show up?
